packages/app/components/Nav/NavBar.tsx

```diff
@@ -98,11 +102,26 @@ const BarsButton = styled.span`
 `;
 
 interface NavBarProps {
-  course: Course;
+  profile: User;
+  courseId: number;
 }
 
-export default function NavBar({ course }: NavBarProps) {
+export default function NavBar({ profile, courseId }: NavBarProps) {
   const [visible, setVisible] = useState<boolean>(false);
+  const [course, setCourse] = useState<CoursePartial>(
+    profile.courses.find((c) => c.course.id === courseId).course
+  );
+  const [queueId, setQueueId] = useState<number>(undefined);
+
+  const { data, error } = useSWR(`api/v1/courses/${courseId}/queue`, async () =>
+    API.course.queues(courseId)
+  );
+
+  useEffect(() => {
```

Comment:
we don't need an effect for this. effects that set state will trigger extraneous re-renders.
just do:
```
const queueId = data && data.length > 0 && data[0].id
```

You can think of `queueId` as "derived" from `data`. It is not itself a piece of the state, it is derived from the `data` state.

Reply:
got it, that makes sense. if `data` were to change, would `queueId` also update?

Reply:
yeah, remember that the entire function component has to execute every time we re-render